packages/cli/src/ceramic-daemon.ts

```diff
@@ -215,89 +143,43 @@ class CeramicDaemon {
     return new CeramicDaemon(ceramic, opts)
   }
 
-  registerAPIPaths (app: core.Express, gateway: boolean): void {
-    app.get(toApiPath('/commits/:docid'), this.commits.bind(this))
-    app.get(toApiPath('/records/:docid'), this.commits.bind(this))
-    app.post(toApiPath('/multiqueries'), this.multiQuery.bind(this))
-    app.get(toApiPath('/documents/:docid'), this.state.bind(this))
-    app.get(toApiPath('/pins/:docid'), this.listPinned.bind(this))
-    app.get(toApiPath('/pins'), this.listPinned.bind(this))
-    app.get(toApiPath('/node/chains'), this.getSupportedChains.bind(this))
-    app.get(toApiPath('/node/healthcheck'), this.healthcheck.bind(this))
+  registerAPIPaths (app: ExpressWithAsync, gateway: boolean): void {
+    app.getAsync(toApiPath('/commits/:docid'), this.commits.bind(this))
```

Comment:
Why do we need to use `ExpressWithAsync`? Doesn't express already handle async routes on its own?

Reply:
It is express 4. It does not handle async routes. https://gist.github.com/ukstv/14377c8ed437aa6243be94d0c532cb25

Express 5 is in alpha.

Reply:
> Starting with Express 5, route handlers and middleware that return a Promise will call next(value) automatically when they reject or throw an error

https://expressjs.com/en/guide/error-handling.html

Reply:
Worth it to upgrade? I'm guessing not but probably worth considering?

Reply:
Vote for ditching Express in favour of [Fastify](https://www.fastify.io), if it ever comes to it.

Yet, it is also worth considering how many breaking or (worse) potentially breaking changes one tries to introduce at a unit of time, and to what effect. I have no idea how that ever-alpha version behaves. Documentation is sparse, mildly speaking. If the goal is to ship a working system, I better stick with the oldie here. As a purely internal detail, it can be upgraded at any time later.

Reply:
Ok great! Agree